src/image.rs

```diff
@@ -47,6 +47,18 @@ pub enum Inode {
     Leaf(Rc<Leaf>),
 }
 
+#[derive(Error, Debug)]
+pub enum ImageError {
+    #[error("Directory entry {0:?} does not exist")]
```

Comment:
Minor style thing but I think where possible it's best to have any user-injected content at the end instead of the middle; it's safer to "visually parse", like:

`#[error("Directory entry does not exist: {0:?}")]`

(Also different tangent on filename display, in bootc I added https://github.com/bootc-dev/bootc/blob/2de8e0d23fb89bed76722ff1466614afacec64b3/utils/src/path.rs#L30 which I think is nicer than the default debug output)

Reply:
To be honest, I'm not a huge fan of how the error reporting on filenames is working generally: most of these functions are going to end up throwing an error for a partial pathname instead of the whole thing, which is kinda weak...  I'll take a todo to clean this up later.